docs/asf-docs/Introduction_hyp3_mintpy.ipynb

```diff
@@ -0,0 +1,527 @@
+    "        coords = all_coords[i]\n",
+    "        ul[0,i] = coords[0]\n",
+    "        lr[0,i] = coords[2]\n",
+    "        ul[1,i] = coords[1]\n",
+    "        lr[1,i] = coords[3]\n",
+    " \n",
+    "        if i == 0:\n",
+    "            xres = all_pixsize[i]\n",
+    "            yres = all_pixsize[i]\n",
+    "\n",
+    "    return ul,lr,xres,yres\n",
+    "\n",
+    "def cutGeotiffsByLine(files):\n",
+    "    ul,lr,xres,yres = getOrigins(files)\n",
+    "    \n",
+    "    diff_ul = np.zeros((2,len(files)))\n",
+    "    diff_ul[0] = (max(ul[0])-ul[0])/xres    \n",
+    "    diff_ul[1] = -1*(min(ul[1])-ul[1])/(-1*yres)\n",
+    "\n",
+    "    print(\"Difference list:\")\n",
+    "    print(diff_ul)\n",
+    "\n",
+    "    lrx = min(lr[0])\n",
+    "    lry = max(lr[1])\n",
+    "    lenx = (lrx-max(ul[0])) / xres\n",
+    "    leny = -1*(lry-min(ul[1])) / (-1*yres)\n",
+    "    if leny < 0:\n",
+    "        leny = abs(leny)\n",
+    "        diff_ul[1] = diff_ul[1] * -1\n",
+    "    print(\"Size of output images {} x {}\".format(lenx, leny))\n",
+    "\n",
+    "    outfiles = []   \n",
+    "    for i in range(len(files)):\n",
+    "        outfile = files[i].replace(\".tif\",\"_clip.tif\")\n",
+    "        print(\"Processing file {} to create file {}\".format(files[i], outfile))\n",
+    "        gdal.Translate(outfile,files[i],srcWin=[diff_ul[0,i],diff_ul[1,i],lenx,leny],noData=0)\n",
+    "        outfiles.append(outfile)\n",
+    "\n",
+    "    return(outfiles)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Clip the files"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get files to be clipped in the outdir and its sub-directories.\n",
+    "files = []\n",
+    "pattern = [\"*.tif\"]                      \n",
+    "for dir,_,_ in os.walk(outdir):\n",
+    "    for item in pattern:\n",
+    "        files.extend(glob.glob(os.path.join(dir, item)))\n",
+    "# clip off files\n",
+    "cutGeotiffsByLine(files)    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2.3.3 Copy metadata file and dem file to DEM subdirectory"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# copy the *.txt and dem.tif in one subdirctory to DEM subdirectory\n",
+    "os.system(\"mkdir -p {}/DEM {}/inputs\".format(outdir, outdir))\n",
+    "prefix = filelist1[-1].split('.zip')[0]\n",
+    "fileprefix = os.path.basename(prefix)\n",
+    "txtfile = \"{dir}/{fileprefix}.txt\".format(dir = prefix, fileprefix = fileprefix)\n",
+    "demfile = \"{dir}/{fileprefix}_dem_clip.tif\".format(dir = prefix, fileprefix = fileprefix)\n",
+    "\n",
+    "cmd1 = \"cp {file} {data_dir}/DEM/dem_clip.txt\".format(file = txtfile, data_dir = outdir)\n",
+    "os.system(cmd1)\n",
```

Comment:
There's no need for `os.system()` in a notebook, Jupyter provides a build-in magic command to execute shell commands:
`! {cmd}`

https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-system

Reply:
changed